.github/workflows/check.yml

```diff
@@ -11,12 +11,10 @@ on:
 jobs:
   check:
     name: Build with gcc and test
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
```

Comment:
This can be a separate commit.

Reply:
Okay, something to keep in mind these changes are interdependent. Glib requires builder bump. But I can split.

Reply:
I have now split it so that CI is updated as first commit and this work is on top of that.